src/main/java/org/opensearch/flowframework/transport/GetWorkflowTransportAction.java

```diff
@@ -79,8 +79,8 @@ protected void doExecute(Task task, WorkflowRequest request, ActionListener<GetW
                         logger.error(errorMessage);
                         listener.onFailure(new FlowFrameworkException(errorMessage, RestStatus.NOT_FOUND));
                     } else {
-                        // Remove any credential from response
-                        Template template = encryptorUtils.redactTemplateCredentials(Template.parse(response.getSourceAsString()));
+                        // Remove any secured field from response
```

Comment:
Can we do this only if the user isn't admin so there is still some way to look at user field, similar to AD we have this check for a similar use case:
https://github.com/opensearch-project/anomaly-detection/blob/f3998e9101d1cd1c246b30756ee7d3a3ca06c757/src/main/java/org/opensearch/timeseries/util/ParseUtils.java#L592

Reply:
Also we still want user for the state index to distinguish who provisions.

Reply:
yes agree but still think only admin should be able to see the user to see who provisioned it, which is what is done in alerting. What do you think about that?

Reply:
> yes agree but still think only admin should be able to see the user to see who provisioned it, which is what is done in alerting.

+1

Reply:
Added the user check for Get API, Search API and Status API.